nifi-nar-bundles/nifi-iotdb-bundle/nifi-iotdb-processors/src/main/java/org/apache/nifi/processors/PutIoTDBRecord.java

```diff
@@ -0,0 +1,303 @@
+            processSession.transfer(flowFile, REL_FAILURE);
+        }
+
+        final boolean aligned = alignedProperty != null ? Boolean.valueOf(alignedProperty) : false;
+        int maxRowNumber = maxRowNumberProperty != null ? Integer.valueOf(maxRowNumberProperty) : 1024;
+
+        try (final InputStream inputStream = processSession.read(flowFile);
+             final RecordReader recordReader =
+                     recordParserFactory.createRecordReader(flowFile, inputStream, getLogger())) {
+            boolean needInitFormatter;
+            IoTDBSchema schema;
+            ValidationResult result;
+
+            result =
+                    schemaProperty != null
+                            ? validateSchemaAttribute(schemaProperty)
+                            : validateSchema(recordReader.getSchema());
+
+            if (!result.getKey()) {
+                getLogger().error("The property `schema` has an error: {}", result.getValue());
+                inputStream.close();
+                recordReader.close();
+                processSession.transfer(flowFile, REL_FAILURE);
+                return;
+            } else {
+                if (result.getValue() != null) {
+                    getLogger().warn("The property `schema` has a warn: {}", result.getValue());
+                }
+            }
+
+            schema =
+                    schemaProperty != null
+                            ? mapper.readValue(schemaProperty, IoTDBSchema.class)
+                            : convertSchema(recordReader.getSchema());
+
+            List<String> fieldNames = schema.getFieldNames(prefix);
+
+            needInitFormatter = schema.getTimeType() != IoTDBSchema.TimeType.LONG;
+
+            HashMap<String, Tablet> tablets = generateTablets(schema, prefix, maxRowNumber);
+            DateTimeFormatter format = null;
+
+            Record record;
+
+            while ((record = recordReader.nextRecord()) != null) {
+                Object[] values = record.getValues();
+                if (format == null && needInitFormatter) {
+                    format = initFormatter((String) values[0]);
+                    if (format == null) {
+                        getLogger().error("{} Record [{}] time format not supported", flowFile, values[0]);
+                        inputStream.close();
+                        recordReader.close();
+                        processSession.transfer(flowFile, REL_FAILURE);
+                        return;
+                    }
+                }
+
+                long timestamp;
+                if (needInitFormatter) {
+                    timestamp = Timestamp
+                            .valueOf(LocalDateTime.parse((String) values[0], format))
+                            .getTime();
+                } else {
+                    timestamp = (Long) values[0];
+                }
+
+                boolean isFulled = false;
```

Comment:
It seems like this should be renamed to `filled`.
```suggestion
                boolean filled = false;
```

Reply:
Refine the code. Thanks for the careful review. @exceptionfactory